app/src/main/java/com/easyfitness/fonte/FontesFragment.java

```diff
@@ -441,7 +441,6 @@ public void onClick(View v) {
         if (hasFocus) {
             switch (v.getId()) {
                 case R.id.editMachine:
-                    machineEdit.setText("");
```

Comment:
If the text is not deleted, then, there is no need to reset the picture just below. 
The picture could be reprocessed when focus is lost for example.

Reply:
I now understand what lines 445 are supposed to do. Will take that into accound.

Reply:
I adapted the code to update the image when focus is gained or when it is lost. 

Ideally, the image should be updated when the exercise type is changed. However, I did not find a nice solution and did not want to touch too many areas. My best idea was something like, e.g., an "workoutValuesInputViewOnChangeListener". On the other hand, cases like these are best handled by a view model in the long run.

Reply:
I agree the need for a view model is getting bigger and bigger every day. 
Anyway, for this version, I will fix this simply. The impact is a bit bigger than expected. I will add a few commit to you pull request that I need to try on an independent branch.